Feature/packages

NickyWeber opened the pull request:
Packages support for cocos2d.

NickyWeber committed:
fullPathForFilename search order changed to go through resolutions first and the search paths.
Packages work with additional search paths. Since packages usually contain only one resolution it may occur that an asset that should be replaced with the asset is found first before the asset is searched. That was the case with the previous search order.

NickyWeber committed:
CCFileUtils: Methods added to find paths of a file in all search paths.
Method added to load all filenameLookups in all search paths and merge them.

NickyWeber committed:
CCSpriteFrameCache: Method added to load all Sprite sheet lookup files in all search paths and merge them in ivar.

NickyWeber committed:
Package classes added.

NickyWeber committed:
SSZipArchive added as submodule.

NickyWeber committed:
SSZipArchive included in project.
Code re-enabled to compile.

NickyWeber committed:
Inline documentation added.

NickyWeber committed:
Merge branch 'develop' into feature/Packages

Conflicts:
	cocos2d-ios.xcodeproj/project.pbxproj
	external/ObjectAL

NickyWeber committed:
Hint added for unzipping delegate callbacks and threads.

slembcke commented:
Err, yes the CCFileUtils issue was my fault. Didn't check compiling for all platforms before pushing.

NickyWeber committed:
Warnings removed: No newline at end of files.

NickyWeber committed:
Partial download resuming more conservative regarding server response.
A server has to explicitly return the Accept-Ranges header with a set value other than none.

NickyWeber committed:
CCPackageHelper added to extract currentOS method from CCPackageManager.
currentOS now a class method.

NickyWeber committed:
CCPackageDownloadManager will allow to resume paused package downloads when enqueue is used.
A failed package download will start over when used with enqueue.

NickyWeber committed:
Compilation error fixed, @class CCPackage added to manager delegate.
Some renaming and comments updated.

NickyWeber committed:
CCPackageManager's API changed.
An instance of a package can now be added to the manager to actually get managed without starting the whole installation life cycle.
Leading to a new method to start the download of a CCPackage instance which has already been added before.
The unzip delegate methods will return on the main thread now.
Some refactoring to attach installData with a method instead of code duplication.

NickyWeber committed:
CCPackageHelper returning a default value instead of nothing to prevent a crash in the cocos2d-test-ios project.
Hopefully this can be solved to get the platform defines set correctly soon.

NickyWeber committed:
First package test added to cocos2d-tests-ios.
Extracted some cocos2d config code in app delegate and made that public to rerun packages tests.

NickyWeber committed:
Packages loaded from bundle, no need for an internet connection anymore.
Test packages added to bundle.

NickyWeber committed:
Warning removed by reordering NSURLProtocolClient method calls.

NickyWeber committed:
Package enabler won't re-add a package if it is already in the search path.

NickyWeber committed:
Safety net added for loadPackage method: Calling it more than once not end in weird states.

NickyWeber committed:
Package Manager's queue can't be set to nil anymore.

NickyWeber committed:
Unit tests added for CCPackageInstallData and CCPackage+InstallData category.
File skeletons for other package classes added.
Moved two methods from CCPackage+InstallData to CCPackageInstallData.

NickyWeber committed:
Refactoring of error constant names to be closer to what they actually mean.

NickyWeber committed:
CCPackageInstaller unit tests added.
Unzipped package added to shared resources for those tests.

NickyWeber committed:
CCUnitTestAssertions.h added with a macro to assert string equality in unit tests.
Simplified unit tests of CCPackageInstallData.
Removed non testable CCPackageHelperTests class.

NickyWeber committed:
Freed CCPackage of CCPackageInstallData code.

NickyWeber committed:
CCPackage unit tests added.
Fix for file urls serialization: Prevent prepending of file:// scheme of installURL for de/serialization cycles.

NickyWeber committed:
CCPackageUnzipper unit tests added.
Password protected archive added for tests. Password is foobar.

NickyWeber committed:
Refactoring: CCPackageInstaller won't enable the package anymore, this is done in the CCPackageManager now.
CCPackageCocos2dEnabler will set relevant status instead of CCPackageManager.

NickyWeber committed:
CCPackageCocos2dEnabler unit tests added.

NickyWeber committed:
Fixed tests for CCTexture. Unit tests for CCPackageCocos2dEnabler broke them.
Solution: Setup cocos2d for each CCTexture test as well as CCPackageCocos2dEnabler tests.

NickyWeber committed:
Bugfix: CCPackageDownlod read wrong fileSize to resume a download.

NickyWeber committed:
CCPackageCocos2dEnabler asset added. Forgot that one in a previous commit.

NickyWeber committed:
CCPackageDownload unit tests for normal request and resuming request(rang request) addded.

NickyWeber committed:
CCPackageDownload unit test added for 404 response.

NickyWeber committed:
CCPackageDownload unit tests finished.

NickyWeber committed:
Physics unit tests fixed by re configuring cocos2d in setUp.

NickyWeber committed:
CCPackageDownloadTests fixed, some nondeterministic errors occured.
Seemed to be a conflict with CCScheduler and the runloop triggering in the unit tests.

NickyWeber committed:
SSZipArchive added as subproject.
Temporarily pointing to https://github.com/NickyWeber/ssziparchive.git as long as the main clone does not contain a static library target.

NickyWeber committed:
SSZipArchive pointing to SB repo fork again.
Updated revision of SSZipArchive.
SSZipArchive header search path added.

NickyWeber committed:
CCPackageDownload will call delegate method if writing file fails.

NickyWeber committed:
CCPackageDownload more tests added.
CCPackageDownload is setting package status for failed downloads instead of download manager.

NickyWeber committed:
CCPackageDownloadManager tests added.
CCPackageDownloadManager header documentation improved for setDownloadPath.
Bugfix for setting a new download path.
Added allDownloads properties. Mainly for testing purposes, it shouldn't hurt.
NSAssert added in CCPackageDownloadManager to assert installData is attached to CCPackageDownload.
CCPackageDownloadManager no longer sets failed download status of a package: CCPackageDownload is doing this now.

NickyWeber committed:
CCPackageDownloadManager: more unit tests added.
Some tidying up of tests.

NickyWeber committed:
Refactoring: Unit tests tidying up.

NickyWeber committed:
Support for default resolution added.
Default resolution is determined by CCFileUtil's searchResolutionsOrder array and is mapped to a sprite builder resolution.
CCPackageManager now offers a simple method to download a package only by providing a name.
CCPackage can now be initialized with a name only.
Some tests added for these cases, although CCPackageHelper tests are disabled at the moment due to a bug in CCFileUtils and handling of searchResolutionsOrder.

NickyWeber committed:
Packages loading and saving moved to CCAppDelegate.

NickyWeber committed:
Published download manager's resumeDownloads flag in CCPackageManager.

NickyWeber committed:
Refactoring: Minor renaming.
Removed double ;.
Some formatting of methods in CCPackageManager's header.

NickyWeber committed:
CCPackage private header added to make properties accessible internally.

NickyWeber committed:
CCPackageUnzipper and tests cleansed of installData.

NickyWeber committed:
CCPackageDownloadManager cleansed of installData.

NickyWeber committed:
CCPackage with install data now.

NickyWeber committed:
CCPackageDownload using private CCPackage header.

NickyWeber committed:
CCPackageCocos2dEnabler using CCPackage private header.

NickyWeber committed:
CCPackageInstaller and tests cleansed of install data.

NickyWeber committed:
CCPackageManager cleansed of install data.

NickyWeber committed:
Install data related files removed from project.
Last import removed in CCPackageTests.

NickyWeber committed:
Merge branch 'develop' into feature/Packages

Conflicts:
	cocos2d-ios.xcodeproj/project.pbxproj
	cocos2d-tests-ios.xcodeproj/project.pbxproj
	cocos2d-ui-tests/ios/AppDelegate.m
	cocos2d/Platforms/iOS/CCAppDelegate.m
	cocos2d/Support/CCFileUtils.h

NickyWeber committed:
Bugfix for CCFileUtils searchResolutionsOrder.
It was internally declared as NSMutableArray but accessible and writable(copy) as NSArray.
This caused all kinds of weird crashes when assigning new values after setup and running the tests.
Reenabled the tests depending on this.
